### PR TITLE
Polishing for App Store

### LINKED ISSUE #458: ExpandedPost displays compact username and community name

### PR DESCRIPTION
**Pull request by EricBAndrews**
# Checklist
- [x] I have read [CONTRIBUTING.md](./CONTRIBUTING.md)
- [x] I have described what this PR contains
- [x] This PR addresses one or more open issues that were assigned to me:
      - closes #458 
- [x] If this PR alters the UI, I have attached pictures/videos

# Pull Request Information

## About this Pull Request
Fixes an issue where the username was not properly displaying in compact mode.

Also takes the opportunity to make the following changes:
- Scaling of the flair is now done using `.imageScale` to make it small in trailing mode
- Flair image is no longer disabled when instance location is `.trailing`
- Avatar URL now always takes the Icon64 URL, since that's what gets prefetched
- Avatar size is now a computed var instead of a function and represented as a `CGSize`
- Instance is now `UIColor.tertiaryLabel` instead of `.secondary` with opacity 0.6

Also made some minor miscellaneous tweaks:
- Fix font size in `.mlemDeemphasized` to be consistent with `.mlem`
- Tab switch animation has been greatly sped up
- Nav bar swipe up is disabled on devices with a home button
- Added haptic feedback to account switcher
- Added 2.5px padding to tab bar when device has a home button (this is the Apple amount as far as I can tell)

Screenshot 2023-08-07 at 6 11 19 PM
Screenshot 2023-08-07 at 6 13 44 PM
Screenshot 2023-08-07 at 6 14 04 PM

### CODE REVIEW

File: Mlem.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2612,7 +2612,7 @@
 					"@executable_path/Frameworks",
 				);
 				MARKETING_VERSION = 1.0.1;
-				PRODUCT_BUNDLE_IDENTIFIER = com.hanners.Mlem;
```
Comment (WestonHanners) [inline Mlem.xcodeproj/project.pbxproj:2615]: nope

File: Mlem/MlemApp.swift [hunk 2]
```diff
@@ -54,6 +55,15 @@ struct MlemApp: App {
                         appearance.backgroundEffect = UIBlurEffect.init(style: .systemThinMaterial)
                         UITabBar.appearance().standardAppearance = appearance
                         UITabBar.appearance().scrollEdgeAppearance = appearance
+                        
+                        // check whether this device has a home button
+                        if #available(iOS 11.0, *),
```
Comment (Sjmarf) [inline Mlem/MlemApp.swift:60]: We only support iOS 16+, right? So we don't need to check whether the user is on iOS 11+.

**Review by Sjmarf — APPROVED** [review]
LGTM, other than one nitpick. I'll approve anyway though, it doesn't matter all that much.

File: Mlem/MlemApp.swift (hunk 2 above)
Comment (EricBAndrews) [reply Mlem/MlemApp.swift]: Fair enough lmao, that's what I get for copying SO :P

**Pull request merged**